diag_manager/fms_diag_object.F90

```diff
@@ -35,7 +35,7 @@ module fms_diag_object_mod
      integer, allocatable, private                    :: diag_id           !< unique id for varable
      class(FmsNetcdfFile_t), dimension (:), pointer   :: fileob => NULL()  !< A pointer to all of the
                                                                            !! file objects for this variable
-     character(len=:), allocatable, dimension(:)      :: metadata          !< metedata for the variable
+     character(len=:), allocatable, dimension(:)      :: metadata          !< metadata for the variable
      logical, private                                 :: static            !< true is this is a static var
      logical, allocatable, private                    :: registered        !< true when registered
```

Comment:
I think its less complex and more efficient if the intrinsic primitives (like logical, integer, real) are not allocatable
and have a default set in the type definiition. E.g. like this
``` 
logical, private    ::  registered => .false.
```
it would help to add functions like ``` logical is_registered()```  (see comments below). Probably,  type fms_dia_object
may nedd one or more initialize functions that the constructor calls.